jsx-global.d.ts

```diff
@@ -8,7 +9,7 @@ declare global {
   namespace JSX {
     type Element = VNode;
     interface IntrinsicElements {
-      [elemName: string]: VNodeData;
+      [elemName: string]: JsxVNodeProps;
     }
   }
 }
```

Comment:
so as a library user you can simply use it as `snabbdom/jsx-global` inside tsconfig rather than having to go into `snabbdom/src/jsx-global`. Just looks neater as a user since jsx pragma itself is imported as `snabbdom/jsx`

Reply:
If it is in `src/jsx-global.d.ts` and it is included in the `files` field in `tsconfig.json`, I expect it to be copied to the root by `tsc` in the CommonJS build. Isn't this so?

Reply:
Not quite. The jsx-global.d.ts here provides type info for input files. That doesn’t get copied. Only the auto generated .d.ts files from input files are outputted in root folder. Basically for outDir compiler option. Tsc makes a .d.ts and a .js for every .ts file.

Reply:
[Right](https://stackoverflow.com/a/56440335/359072). How about rename it to `src/jsx-global.ts` as suggested in that answer?

Reply:
I’m not sure what you mean. Do you want the import to still be `snabbdom/src/jsx-global` ?

Reply:
If it is renamed to a `.ts` file, `src/jsx-global.ts` then it will be compiled and end up in the two build outputs. Just make sure it's in the `tsconfig.json` `files` field. Do you like this idea?

Reply:
if we do that, then tsc will generate an empty jsx-global.js, and we'll get many copies of jsx-global.d.ts. One in root, one in /es and one in src. Feels uglier. Honestly I don't have strong opinions. If you feel it should really be `src/jsx-global.d.ts` I can revert those changes and move it back again.

Reply:
I appreciate your willingness to change according toy preference. But let me explain why I prefer it that way.

It seems simpler if this file is considered source code and lives with all the other source code. There is no "Oh, and there's this other file outside of `src`. That feels like less complexity to me.

Regarding the multiple useless build artifacts I don't feel that's an issue at all. Just ignore them in git and include in the package only what should be. It's a whitelist anyway.